api/src/routes/product-list.ts

```diff
@@ -21,7 +26,7 @@ export async function productListHandler(req: Request, res: Response) {
         shopId: true,
       },
       with: { category: { columns: { id: true, name: true } } },
-      where: (products, { eq }) => eq(products.shopId, user.shopId),
+      where: (products, { eq }) => eq(products.shopId, shopId),
```

Comment:
Same as above. We only need the shop id for querying, so we can use `getShopIdBySite` instead to simplify things.